tests/Feature/Api/V1/UserProfileTest.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Feature\Api\V1;
+
+use App\Site\Models\User;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Tests\TestCase;
+
+class UserProfileTest extends TestCase
+{
+    use RefreshDatabase;
+    use BootstrapsApiV1;
+
+    public function testItValidates(): void
+    {
+        $this->get($this->apiUrl('GetUserProfile'))
+            ->assertJsonValidationErrors([
+                'u',
+            ]);
+    }
+
+    public function testGetUserProfileUnknownUser(): void
+    {
+        $this->get($this->apiUrl('GetUserProfile', ['u' => 'nonExistant']))
+            ->assertNotFound()
+            ->assertJson([]);
+    }
+
+    public function testGetUserProfile(): void
+    {
+        /** @var User $user */
+        $user = User::factory()->create();
+
+        $this->get($this->apiUrl('GetUserProfile', ['u' => $user->User]))
+            ->assertSuccessful()
+            ->assertJson([
+                'User' => $user->User,
+                'UserPic' => sprintf("/UserPic/%s.png", $user->User),
+                'MemberSince' => $user->Created?->__toString(),
+                'RichPresenceMsg' => empty($user->RichPresenceMsg) || $user->RichPresenceMsg === 'Unknown' ? null : $user->RichPresenceMsg,
```

Comment:
Duplicating the logic in the unit test doesn't really validate the expectations. You should have an explicit expectation of `$user->richPresenceMsg` or `null`.